crates/sui-indexer/src/db.rs

```diff
@@ -128,13 +135,63 @@ pub fn get_pool_connection(pool: &ConnectionPool) -> Result<PoolConnection, Inde
     })
 }
 
+#[derive(QueryableByName)]
+pub struct StoredMigration {
+    #[diesel(sql_type = diesel::sql_types::Text)]
+    pub version: String,
+}
+
+pub async fn check_db_migration_consistency(pool: ConnectionPool) -> Result<(), IndexerError> {
+    info!("Starting compatibility check");
+    let query = "SELECT version FROM __diesel_schema_migrations ORDER BY version";
+    let db_migrations: Vec<_> = run_query_async!(&pool, move |conn| {
+        sql_query(query).load::<StoredMigration>(conn)
+    })?
+    .into_iter()
+    .map(|m| m.version)
+    .collect();
+    info!("Migration Records from the DB: {:?}", db_migrations);
+
+    let known_migrations: Vec<_> = MIGRATIONS
+        .migrations()
+        .unwrap()
+        .into_iter()
+        .map(|m: Box<dyn Migration<Pg>>| {
+            let full_name = m.name().to_string().replace("-", "");
+            let first_non_digit = full_name
+                .find(|c: char| !c.is_ascii_digit())
+                .unwrap_or(full_name.len());
+            full_name[..first_non_digit].to_string()
+        })
+        .collect();
+    info!(
+        "Migration Records from local schema: {:?}",
+        known_migrations
+    );
+    for i in 0..max(known_migrations.len(), db_migrations.len()) {
+        let local_migration_record = known_migrations.get(i).cloned().unwrap_or_default();
+        let db_migration_record = db_migrations.get(i).cloned().unwrap_or_default();
```

Comment:
when a new migration is added in a pr, `local_migration_record` will have an incremental migration, when is the new migration going to be applied to `db_migration_record`?

Reply:
My understanding is that as part of our release process, someone would start a new indexer instance that contains the new schema with the new migration scripts and backfill it when we want to release a new indexer with schema changes.